src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1474,6 +1484,68 @@ protected void dropEquipment(final boolean par1, final int par2)
         }
     }
 
+    public void trySleep(BlockPos bedLocation)
+    {
+        final IBlockState state = this.world.isBlockLoaded(bedLocation) ? this.world.getBlockState(bedLocation) : null;
+        final boolean isBed = state != null && state.getBlock().isBed(state, this.world, bedLocation, this);
+
+        if (!isBed)
+            return;
+
+        this.setPosition((double)((float)bedLocation.getX() + CONST_HALF_BLOCK), (double)((float)bedLocation.getY() + CONST_BED_HEIGHT), (double)((float)bedLocation.getZ() + CONST_HALF_BLOCK));
+
+        this.motionX = 0.0D;
+        this.motionY = 0.0D;
+        this.motionZ = 0.0D;
+
+        setIsAsleep(true);
+
+        dataManager.set(DATA_BED_POS, bedLocation);
+    }
+
+    public BlockPos getBedLocation()
+    {
+        return dataManager.get(DATA_BED_POS);
+    }
+
+    public float getRenderOffsetX()
+    {
+        if (!isAsleep())
+        {
+            return 0f;
+        }
+
+        final IBlockState state = this.world.isBlockLoaded(getBedLocation()) ? this.world.getBlockState(getBedLocation()) : null;
+        final boolean isBed = state != null && state.getBlock().isBed(state, this.world, getBedLocation(), this);
+        final EnumFacing enumfacing = isBed && state.getBlock() instanceof BlockHorizontal ? state.getValue(BlockHorizontal.FACING) : null;
+
+        if (enumfacing == null)
+        {
+            return 0f;
+        }
+
+        return -1.5F * (float)enumfacing.getFrontOffsetX();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1.5F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)